src/art.rs

```diff
@@ -1221,54 +1162,41 @@ impl<P: KeyTrait, V: Clone> Tree<P, V> {
         self.insert_common_mut(key, value, version, ts, false, true)
     }
 
-    pub(crate) fn remove_from_node(
-        node: Option<&Arc<Node<P, V>>>,
+    fn remove_recursive(
+        cur_node: &Node<P, V>,
         key: &P,
+        depth: usize,
     ) -> (Option<Arc<Node<P, V>>>, bool) {
-        // Directly match on the root to simplify logic
-        let (new_root, is_deleted) = match &node {
-            Some(root) if !root.is_twig() => {
-                // Determine the prefix of the key after the current depth.
-                let key_prefix = key.prefix_after(0);
-                // Find the longest common prefix between the current node's prefix and the key's prefix.
-                let longest_common_prefix = root.prefix().longest_common_prefix(key_prefix);
-
-                if longest_common_prefix != root.prefix().len() {
-                    // If the longest common prefix doesn't match the root's prefix length, no deletion occurs.
-                    (Some(Arc::clone(root)), false)
-                } else {
-                    // Proceed with recursive removal if the prefixes match.
-                    let (new_root, removed) =
-                        Node::remove_recurse(root, key, longest_common_prefix);
-
-                    if removed
-                        && new_root.as_ref().unwrap().is_inner()
-                        && new_root.as_ref().unwrap().num_children() == 0
-                    {
-                        // If the node was deleted and it has no children, return None as the new root.
-                        (None, removed)
-                    } else {
-                        (new_root, removed)
-                    }
+        let key_prefix = key.prefix_after(depth);
```

Comment:
Can we keep all the comments intact? Would be helpful later on

Reply:
Updated the comments.